pyomo/contrib/parmest/parmest.py

```diff
@@ -229,11 +229,119 @@ def _experiment_instance_creation_callback(
 
 def SSE(model):
     """
-    Sum of squared error between `experiment_output` model and data values
+    Sum of squared error between the model prediction of measured variables and data values,
+    assuming Gaussian i.i.d errors.
     """
-    expr = sum((y - y_hat) ** 2 for y, y_hat in model.experiment_outputs.items())
+    expr = sum((y - y_hat) ** 2 for y_hat, y in model.experiment_outputs.items())
     return expr
 
+def SSE_weighted(model):
+    """
+    Weighted sum of squared error between the model prediction of measured variables and data values,
+    assuming Gaussian i.i.d errors.
+    """
+    expr = (1 / 2) * sum(
+        ((y - y_hat) / model.measurement_error[y_hat]) ** 2
+        for y_hat, y in model.experiment_outputs.items()
+    )
+    return expr
+
+# Calculate the sensitivity of measured variables to parameters using central finite difference
+def _compute_jacobian(model, relative_perturbation, solver_option="ipopt"):
+    """
+    Computes the Jacobian matrix using central finite difference scheme
+
+    Arguments:
+        model: Pyomo model containing experiment_outputs and measurement_error
+        relative_perturbation: value used to perturb the objectives
+
+    Returns:
+        J: Jacobian matrix
+    """
+    # get the measured variables
+    y_hat_list = [y_hat for y_hat, y in model.experiment_outputs.items()]
+
+    # get the parameters
+    params = [k for k, v in model.unknown_parameters.items()]
+    param_values = [p.value for p in params]
+
+    # get the number of parameters and measured variables
+    n_params = len(param_values)
+    n_outputs = len(y_hat_list)
+
+    # compute the sensitivity of measured variables to the parameters (Jacobian)
+    J = np.zeros((n_outputs, n_params))
+
+    for i, param in enumerate(params):
+        # store original value of the parameter
+        orig_value = param_values[i]
+
+        # Forward perturbation
+        param.fix(orig_value + relative_perturbation)
+
+        # solve model
+        solver = pyo.SolverFactory(solver_option)
+        solver.solve(model)
+
+        # forward perturbation measured variables
+        y_hat_plus = [pyo.value(y_hat) for y_hat, y in model.experiment_outputs.items()]
+
+        # Backward perturbation
+        param.fix(orig_value - relative_perturbation)
+
+        # resolve model
+        solver.solve(model)
+
+        # backward perturbation measured variables
+        y_hat_minus = [pyo.value(y_hat) for y_hat, y in model.experiment_outputs.items()]
+
+        # Restore original parameter value
+        param.fix(orig_value)
+
+        # Central difference approximation for the Jacobian
+        J[:, i] = [(y_hat_plus[w] - y_hat_minus[w]) / (2 * relative_perturbation) for w in range(len(y_hat_plus))]
+
+    return J
+
+# compute the Fisher information matrix of the estimated parameters
+def compute_FIM(model, relative_perturbation, solver_option="ipopt"):
```

Comment:
Allow the user to provide a solver object instead of the solver name. Likewise, allow the user to select from several methods including: (1) central finite difference, (2) kaug, and (3) reduced Hessian.